internal/email/email.go

```diff
@@ -61,56 +99,88 @@ func NewMessage(
-		TermsURL         string
-		PrivacyURL       string
-		SignInURL        string
-		RoleName         string
-		RoleVerb         string
-	}{
-		AdminName:        sponsor,
-		OrganizationName: project,
-		InvitationURL:    inviteURL,
-		RecipientEmail:   inviteeEmail,
-		MinderURL:        minderURL,
-		TermsURL:         DefaultMinderTermsURL,
-		PrivacyURL:       DefaultMinderPrivacyURL,
-		SignInURL:        minderURL,
-		RoleName:         role,
-		RoleVerb:         authz.AllRolesVerbs[authz.Role(role)],
+// getHTMLBodyString returns the HTML body for the email based on the message payload.
+// If there is an error creating the HTML body, it will try to return the text body instead
+func getEmailBodyHTML(ctx context.Context, data bodyData) string {
+	var b strings.Builder
+	bHTML := bodyHTML
+
+	bodyHTMLTmpl, err := util.NewSafeHTMLTemplate(&bHTML, "body-invite-html")
+	if err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error creating the HTML template for email invitations")
+		return getEmailBodyText(ctx, data)
+	}
+	err = bodyHTMLTmpl.Execute(ctx, &b, data, EmailBodyMaxLength)
+	if err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error executing the HTML template for email invitations")
+		return getEmailBodyText(ctx, data)
 	}
+	return b.String()
+}
 
-	// TODO: Load the email template from elsewhere
+// getTextBodyString returns the text body for the email based on the message payload
+func getEmailBodyText(ctx context.Context, data bodyData) string {
+	var b strings.Builder
+	bText := bodyText
 
-	// Parse the template
-	tmpl, err := templates.Parse(bodyHTML)
+	bodyTextTmpl, err := util.NewSafeHTMLTemplate(&bText, "body-invite-text")
 	if err != nil {
-		zerolog.Ctx(ctx).Error().Err(err).Msg("error parsing the HTML template for email invitations")
-		// Default to the text body
-		return getEmailBodyText(inviteURL, sponsor, project, role)
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error creating the text template for email invitations")
+		return ""
 	}
-	// Execute the template
-	var b strings.Builder
-	if err := tmpl.Execute(&b, data); err != nil {
+	err = bodyTextTmpl.Execute(ctx, &b, data, EmailBodyMaxLength)
+	if err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error executing the text template for email invitations")
 		return ""
 	}
 	return b.String()
 }
 
-// getEmailBodyText returns the text body for the email based on the message payload
-func getEmailBodyText(inviteURL, sponsor, project, role string) string {
-	return fmt.Sprintf("You have been invited to join %s as %s by %s. Visit %s to accept the invitation.",
-		project, role, sponsor, inviteURL)
-}
-
 // getEmailSubject returns the subject for the email based on the message payload
 func getEmailSubject(project string) string {
 	return fmt.Sprintf("You have been invited to join the %s organization in Minder", project)
 }
+
+// isValidField checks if the string is empty or contains HTML or JavaScript injection
+func isValidField(str string) error {
```

Comment:
I think this is also covered by html/template -- do we need both?

Reply:
I decided to add it because html/template was not erroring if it finds html in the template input, instead it escapes it and so we end up with things like `&lt;script&gt;`

Reply:
Can you add a comment that we want to detect and error, rather than escape?